File: awesome/helpers/getWeather.lua
#!/bin/lua

local function readFile(path)
	local f = io.open(path)
	local output = {}
	local index = 0
	for line in f:lines() do
		output[index] = line
		index = index + 1
	end
	return output
end

local function getWeather()
	local json = require("lunajson")
	local ltn12 = require("ltn12")
	local path = string.format("%s/.config/awesome/open_weather", os.getenv("HOME"))
	local credentials = readFile(path)
	local http = require("socket.http")
	local body = {}
	local res, code, headers, status = http.request({
		url = "https://api.openweathermap.org/data/2.5/weather?id=" .. credentials[0] .. "&appid=" .. credentials[1],
		sink = ltn12.sink.table(body),
	})
	local response = table.concat(body)
	local content = json.decode(response)
	local temperature = math.floor((content.main.temp - 273.15) + 0.5)
	local description = content.weather[1].main

	print(temperature)
	print(description)
end

getWeather()

File: awesome/helpers/get_weather.py
#!/bin/python

import requests
import json
import math
import os


def read_credential(path):
    with open(path) as file:
        content = file.readlines()
    credentials = {"key": content[0].strip("\n"), "city": content[1].strip("\n")}
    return credentials


if __name__ == "__main__":
    HOME = os.environ["HOME"]
    PATH = f"{HOME}/.config/awesome/open_weather"
    CREDENTIALS = read_credential(PATH)
    URL = f"https://api.openweathermap.org/data/2.5/weather?id={CREDENTIALS['city']}&appid={CREDENTIALS['key']}"

    response = json.loads(requests.get(URL).content)
    temperature = math.floor((response["main"]["temp"] - 273.15) + 0.5)
    description = response["weather"][0]["main"]

    print(temperature, description)
